optuna/trial.py

```diff
@@ -388,14 +388,13 @@ def should_prune(self, step=None):
             A boolean value. If :obj:`True`, the trial should be pruned. Otherwise, the trial will
             be continued.
         """
-        if step is None:
-            step = max(self.storage.get_trial(self._trial_id).intermediate_values.keys())
-        else:
+        if step is not None:
             warnings.warn(
                 'The use of `step` argument is deprecated. '
                 'You can omit to pass this parameter.', DeprecationWarning)
```

Comment:
This message seems a bit inconsistent with the current behavior because `step` argument is no longer used. I think it is helpful for users if we explain that the last step is used instead of the step given by the argument.
What do you think of it?

Reply:
Thank you for reviewing! I modified the warning message at 0864970.